content/en/docs/migrating/_index.md

```diff
@@ -829,6 +899,355 @@ to wait for a {{% alias service=S3 %}} Bucket to exist, you must construct a `Bu
+    })
+}
+
+func main() {
+    cfg, err := config.LoadDefaultConfig(context.Background())
+    if err != nil {
+        // ...
+    }
+
+    svc := s3.NewFromConfig(cfg, WithReadPutObjectOutput)
+    // ...
+}
+```
+
+### HTTP request/response
+
+The `HTTPRequest` and `HTTPResponse` fields from `Request` are now exposed in
+specific middleware phases. Since middleware is transport-agnostic, you must
+perform a type assertion on the middleware input or output to reveal the
+underlying HTTP request or response.
+
+Request handlers which reference `Request.HTTPRequest` and
+`Request.HTTPResponse` must be migrated to middleware.
+
+#### migrating `HTTPRequest`
+
+```go
+// V1
+
+import "github.com/aws/aws-sdk-go/request"
+
+func withHeader(header, val string) request.Option {
+    return func(r *request.Request) {
+        request.HTTPRequest.Header.Set(header, val)
+    }
+}
+
+func main() {
+    sess := session.Must(session.NewSession())
+    sess.Handlers.Build.PushBack(withHeader("x-user-header", "..."))
+
+    svc := s3.New(sess)
+    // ...
+}
+```
+
+```go
+// V2
+
+import (
+    "context"
+    "fmt"
+
+    "github.com/aws/aws-sdk-go-v2/config"
+    "github.com/aws/aws-sdk-go-v2/service/s3"
+    "github.com/aws/smithy-go/middleware"
+    smithyhttp "github.com/aws/smithy-go/transport/http"
+)
+
+type withHeader struct {
+    header, val string
+}
+
+// implements middleware.BuildMiddleware, which runs AFTER a request has been
+// serialized and can operate on the transport request
+var _ middleware.BuildMiddleware = (*withHeader)(nil)
+
+func (*withHeader) ID() string {
+    return "withHeader"
+}
+
+func (m *withHeader) HandleBuild(ctx context.Context, in middleware.BuildInput, next middleware.BuildHandler) (
+    out middleware.BuildOutput, metadata middleware.Metadata, err error,
+) {
+    req, ok := in.Request.(*smithyhttp.Request)
+    if !ok {
+        return out, metadata, fmt.Errorf("unrecognized transport type %T", in.Request)
+    }
+
+    req.Header.Set(m.header, m.val)
+    return next.HandleBuild(ctx, in)
+}
+
+func WithHeader(header, val string) func (*s3.Options) {
+    return func(o *s3.Options) {
+        o.APIOptions = append(o.APIOptions, func (s *middleware.Stack) error {
+            return s.Build.Add(&withHeader{
+                header: header,
+                val: val,
+            }, middleware.After)
+        })
+    }
+}
+
+func main() {
+    cfg, err := config.LoadDefaultConfig(context.Background())
+    if err != nil {
+        // handle error
+    }
+
+    svc := s3.NewFromConfig(cfg, WithHeader("x-user-header", "..."))
+    // ...
+}
+```
+
+### Handler phases
+
+SDK v2 middleware phases are the successor to v1 handler phases.
+
+The following table provides a rough mapping of v1 handler phases to their
+equivalent location within the V2 middleware stack:
+
+| v1 handler name | v2 middleware phase |
+| --------------- | ------------------- |
+| Validate | Initialize |
+| Build | Serialize |
```

Comment:
does V2's `Build` not have an equivalent in V1?

Reply:
It doesn't, `Build` and `Serialize` as we know them in v2 were one thing `Build` in v1